### PR TITLE
add "main", "module" and "types" field

### PR DESCRIPTION
**Pull request by rifler**
fixes https://github.com/sindresorhus/ky/issues/283

### CODE REVIEW

**Comment by sholladay** [comment]
CI failed. You can run the tests locally with `npm test`.

**Comment by rifler** [comment]
> CI failed. You can run the tests locally with `npm test`.

Fixed, thanks!

But this test fails for me in `master` branch too, I think someone merge PR without passing CI tests

**Comment by sholladay** [comment]
No, that error message comes from the `node-fetch` dependency. They probably changed it in a minor or patch version since the last time our CI ran. I'd argue that they broke SemVer rules. Ideally, they would roll back that change and release it as SemVer major. But oh well. Since it only affects our tests, I won't bother them about it.

**Comment by rifler** [comment]
Oh, I see

maybe we should use package-lock.json for that kind of situations?
Builds must be reproducible over time

I can add package-lock in this PR

**Comment by sholladay** [comment]
Sindre and I prefer to use lockfiles in apps but not in reusable libraries. There's a lot of discussion about it on Sindre's AMA if you want to dig into that more. Anyway, thanks for fixing it.

**Comment by rifler** [comment]
> Sindre and I prefer to use lockfiles in apps but not in reusable libraries. There's a lot of discussion about it on Sindre's AMA if you want to dig into that more. Anyway, thanks for fixing it.

npm never publish lock file (and doesn't use it in dependency).

lock files are nessesary for libraries for reproducible build in CI and so, to prevent situations like this - test fails, even if I didn't touch any code

**Comment by sholladay** [comment]
The short of it is that for libraries we prefer for CI to test the latest dependencies. After all, if Ky were to use a lockfile, It would only affect our CI, not end-users' apps. Thus the lockfile actually provides very little value for a library and can actually cause CI to be green even though a dependency had broken that would affect anyone installing Ky from scratch, which makes the lockfile actively harmful.

Lockfiles for apps, not libraries.

**Comment by sindresorhus** [comment]
> Lockfiles for apps, not libraries.

Yup. https://github.com/sindresorhus/ama/issues/479#issuecomment-310661514

File: package.json [hunk 1]
```diff
@@ -5,6 +5,11 @@
 	"license": "MIT",
 	"repository": "sindresorhus/ky",
 	"funding": "https://github.com/sindresorhus/ky?sponsor=1",
+	"main": "./umd.js",
+	"module": "./index.js",
+	"browser": "./index.js",
+	"unpkg": "./index.js",
+	"types": "./index.d.ts",
```
Comment (sindresorhus) [inline package.json:12]: This is not needed. TS infers this automatically when named `index.d.ts`.

File: package.json [hunk 2]
```diff
@@ -5,6 +5,11 @@
 	"license": "MIT",
 	"repository": "sindresorhus/ky",
 	"funding": "https://github.com/sindresorhus/ky?sponsor=1",
+	"main": "./umd.js",
```
Comment (sindresorhus) [inline package.json:8]: This will definitely break all the things. I think https://github.com/sindresorhus/ky/pull/244/files is a much better solution.

Comment (rifler) [reply package.json:8]: Ok! will it work for webpack 4 for example?

Comment (rifler) [reply package.json:8]: why it will break all the things?

File: package.json (hunk 1 above)
Comment (rifler) [reply package.json:12]: Ok, I thought that explicit it better than implicit

**Comment by sholladay** [comment]
The issue with the browser tests has been fixed on the master branch.

**Comment by sindresorhus** [comment]
Closing as this is not going anywhere.

File: package.json (hunk 2 above)
Comment (sindresorhus) [reply package.json:8]: Because `ky` is an ESM package so the "main" export should be just that.

**Pull request closed without merging**